fork-off/src/fetching.rs

```diff
@@ -37,10 +37,22 @@ impl StateFetcher {
                 .await
                 .unwrap();
 
+            let child_storage_map_res = self
+                .client
+                .get_child_storage_for_key(key.clone(), &block)
+                .await;
```

Comment:
Above if `self.client.get_storage` returns error (`RpcError`) we call `unwrap()` and short-circuit the loop.

Reply:
Error here means that we passed an incorrect storage key: since it is a correct key in general, we know that there is just no child trie here (now I realize that we actually do not know that - error can probably have a different cause, will check that)

Reply:
If there is no child trie there we should be able to recognize that differently. Maybe change the return type to `Result<Option<_>, RpcError>` and handle it on the caller's site adequately.

Reply:
Sounds right

Reply:
If you do that, we can `unwrap()` here as well (or just `.await?`) since the only errors that would be returned are RPC errors (connection etc.) since the _child storage does not exist_ (valid case) would be represented as `None` and be handled properly below.

Reply:
I think it's done

Reply:
Yes, it is. If @DamianStraszak approved I guess it means that works :)